### PR TITLE
HenryJk

### PR DESCRIPTION
**Pull request by HenryJk**

### CODE REVIEW

**Comment by robertvictoreen** [comment]
Great storage model for Assignment and Mark. This will enable integration with the respective commands.

**Comment by HenryJk** [comment]
Sort of, but it is kind of hard to do relational database stuffs. Like when you want to swap assignment order, you must swap it for every person, so I just decided to use UniqueId+HashMap to do that.

**Comment by robertvictoreen** [comment]
Yes, a relational database would have been ideal but I believe this is sufficient. However, for the MarkCommand it will need to identify which assignment to record the mark. Also, now we need an AddAssignment command.

File: src/main/java/seedu/address/model/Model.java [hunk 1]
```diff
@@ -42,6 +43,33 @@
      */
     void updatePerson(Person target, Person editedPerson);
 
+    /**
+     * Returns true if a person with the same identity as {@code person} exists in the address book.
```
[the remark this answers is not in the document]
Comment (HenryJk) [reply src/main/java/seedu/address/model/Model.java]: Fixed

File: src/main/java/seedu/address/model/assignment/Assignment.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
+package seedu.address.model.assignment;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.commons.util.UniqueIdUtil.createUniqueId;
+
+import java.util.Objects;
+/**
+ * Represents a Person in the address book.
```
[the remark this answers is not in the document]
Comment (HenryJk) [reply src/main/java/seedu/address/model/assignment/Assignment.java]: Fixed

File: src/main/java/seedu/address/model/assignment/UniqueAssignmentList.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.assignment.exceptions.AssignmentNotFoundException;
+import seedu.address.model.assignment.exceptions.DuplicateAssignmentException;
+
+/**
```
[the remark this answers is not in the document]
Comment (HenryJk) [reply src/main/java/seedu/address/model/assignment/UniqueAssignmentList.java]: Fixed

**Comment by spencertan96** [comment]
Which is the UI supposed to keep track of uh? FilteredAssignmentList or assignments in AddressBook.java

**Comment by HenryJk** [comment]
I think it should be FilteredAssignmentList() in the model, but I will check on it tomorrow

**Pull request merged**